force-app/main/default/classes/ADDR_Addresses_UTIL_TEST.cls

```diff
@@ -0,0 +1,1341 @@
+    }
+
+    /**************************************************************************************************************************
+     ****************************************************** isContactAddressChanged *******************************************
+     **************************************************************************************************************************/
+    @isTest
+    static void testIsContactAddressChangedBothNull() {
+        Test.startTest();
+        Boolean isContactAddressChanged = ADDR_Addresses_UTIL.isContactAddressChanged(null, null);
+        Test.stopTest();
+
+        System.assertEquals(false, isContactAddressChanged);
+    }
+
+    @isTest
+    static void testIsContactAddressChangedCon1Null() {
+        Contact con2 = UTIL_UnitTestData_TEST.getContact();
+        con2.MailingStreet = 'Some Street';
+
+        Test.startTest();
+        Boolean isContactAddressChanged = ADDR_Addresses_UTIL.isContactAddressChanged(null, con2);
+        Test.stopTest();
+
+        System.assertEquals(true, isContactAddressChanged);
+    }
+
+    @isTest
+    static void testIsContactAddressChangedCon1NullCon2EmptyTypeEmpty() {
+        Contact con2 = UTIL_UnitTestData_TEST.getContact();
+
+        Test.startTest();
+        Boolean isContactAddressChanged = ADDR_Addresses_UTIL.isContactAddressChanged(null, con2);
+        Test.stopTest();
+
+        System.assertEquals(false, isContactAddressChanged);
+    }
+
+    @isTest
+    static void testIsContactAddressChangedCon2NullCon1NotEmptyTypeNotEmpty() {
+        Contact con1 = UTIL_UnitTestData_TEST.getContact();
+        con1.Primary_Address_Type__c = 'Some Type';
+
+        Test.startTest();
+        Boolean isContactAddressChanged = ADDR_Addresses_UTIL.isContactAddressChanged(con1, null);
+        Test.stopTest();
+
+        System.assertEquals(true, isContactAddressChanged);
+    }
+
+    @isTest
+    static void testIsContactAddressChangedBothNotNullSameAddressInfo() {
+        Contact con1 = UTIL_UnitTestData_TEST.getContact();
+
+        Test.startTest();
+        Boolean isContactAddressChanged = ADDR_Addresses_UTIL.isContactAddressChanged(con1, null);
+        Test.stopTest();
+
+        System.assertEquals(false, isContactAddressChanged);
+    }
+
+    @isTest
+    static void testIsContactAddressChangedBothNotNullDifferentAddressInfo() {
+        Contact con1 = UTIL_UnitTestData_TEST.getContact();
+        Contact con2 = UTIL_UnitTestData_TEST.getContact();
+        con1.MailingStreet = 'Some Street';
+
+        Test.startTest();
+        Boolean isContactAddressChanged = ADDR_Addresses_UTIL.isContactAddressChanged(con1, con2);
+        Test.stopTest();
+
+        System.assertEquals(true, isContactAddressChanged);
+    }
+
+    @isTest
+    static void testIsContactAddressChanged() {
+        /* return (!isContactAddressEmpty(con1) || String.isNotBlank(con1.Primary_Address_Type__c)); does not make any sense
```

Comment:
This is a distinct use case for test coverage. So we have 4 use cases for con1 or con2 being null:

1. `isContactAddressEmpty(con1)` returns true
2. `isContactAddressEmpty(con1)` returns false
3. `String.isNotBlank(con1.Primary_Address_Type__c)` returns true
4. `String.isNotBlank(con1.Primary_Address_Type__c)` returns false

So explicitly cover these scenarios.

Reply:
Yeah this is a mistake that i forgot to remove the test method. All the scenarios you mentioned were covered by test methods above.